starknet_py/net/client_models.py

```diff
@@ -335,6 +337,15 @@ class ContractClass:
     abi: Optional[AbiDictList] = None
 
 
+@dataclass
+class CompiledContract(ContractClass):
+    """
+    Dataclass representing ContractClass with required abi.
+    """
+
+    abi: AbiDictList = field(default_factory=list)
```

Comment:
I think we could add some sort of post load verification to assert that abi is not an empty list, otherwise it's an okay approach.

Reply:
I don't think we need to. `abi` is a required key in the schema, and its type is `list`. So the `default_factory=list` actually does nothing. It's just a fix for keyword arguments before positional ones.